src/OrleansAzureUtils/AzureConfigUtils.cs

```diff
@@ -181,16 +181,20 @@ private static IEnumerable<DirectoryInfo> FindAppDirectoryLocations()
         /// <summary>
         /// Get the instance named for the specified Azure role instance
         /// </summary>
-        /// <param name="deploymentId">Azure Deployment Id for this service</param>
         /// <param name="roleInstance">Azure role instance information</param>
         /// <returns>Instance name for this role</returns>
-        public static string GetInstanceName(string deploymentId, RoleInstance roleInstance)
+        public static string GetInstanceName(RoleInstance roleInstance)
         {
+            // Try to remove the deploymentId part of the instanceId, if it is there.
+            // The goal is mostly to remove undesired characters that are being added to deploymentId when run in Azure emulator.
+            // Notice that we should use RoleEnvironment.DeploymentId and not the deploymentId that is being passed to us by the user, 
+            // since in general this may be an arbitrary string and thus we will not remove the RoleEnvironment.DeploymentId correctly.
+            string deploymentId = RoleEnvironment.DeploymentId;
```

Comment:
This is the essence of this change. Use the actual RoleEnvironment.DeploymentId that is used by the emulator and not what user may pass as a deployment id (which could be an arbitrary string).